src/app/loans/loans-view/loan-account-actions/approve-loan/approve-loan.component.ts

```diff
@@ -0,0 +1,180 @@
+/** Angular Imports. */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators, FormControl } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { distinctUntilChanged } from 'rxjs/operators';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services. */
+import { LoansService } from 'app/loans/loans.service';
+
+/**
+ * Approve Loan component.
+ */
+@Component({
+  selector: 'mifosx-approve-loan',
+  templateUrl: './approve-loan.component.html',
+  styleUrls: ['./approve-loan.component.scss']
+})
+export class ApproveLoanComponent implements OnInit {
+
+  /** Approve Loan form. */
+  approveLoanForm: FormGroup;
+  /** Loan data. */
+  loanData: any = new Object();
+  /** Minimum Date allowed. */
+  minDate = new Date(2000, 0, 1);
+  /** Approved amount. */
+  approvedAmount: any;
+  /** Disbursement details. */
+  disbursementDetails: any[] = new Array();
+  /** Disbursement count. */
+  disbursementCount = 0;
+  /** Total Amount. */
+  total = 0;
+
+  /**
+   * Retrieve data from `Resolver`.
+   * @param formBuilder Form Builder.
+   * @param route Activated Route.
+   * @param datePipe Date Pipe.
+   * @param loanService Loan Service.
+   * @param router Router.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private datePipe: DatePipe,
+              private loanService: LoansService,
+              private router: Router) {
+    this.route.data.subscribe((data: { loanData: any, loanTemplate: any}) => {
+      this.loanData.loanData = data.loanData;
+      this.loanData.template = data.loanTemplate;
+      this.approvedAmount = this.loanData.template.approvalAmount;
```

Comment:
As the data hasn't be passed while routing, this will throw an error. And the particular routing isn't linked directly(i.e. there is a generic linking) you need to include the resolve field in the `:action` routing and receive the data in the `loan-account-actions.component.ts` file. After that based on the sub-component(the different action button components) which requires the data, you can share them params. You can refer to the [link](https://stackoverflow.com/questions/45408482/how-to-pass-multiple-data-to-child-component-using-input)  to get  a view on how to link the data.

Also you can search for `@Input()` to know more about this.

Reply:
Please check. I have done some changes